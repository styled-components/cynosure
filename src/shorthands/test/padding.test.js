// @flow
import padding from '../padding'

describe('padding', () => {
  it('properly applies a value when passed only one', () => {
    expect(padding('12px')).toEqual({
      paddingBottom: '12px',
      paddingLeft: '12px',
      paddingRight: '12px',
      paddingTop: '12px',
    })
  })
  it('properly applies values when passed two', () => {
    expect(padding('12px', '24px')).toEqual({
      paddingBottom: '12px',
      paddingLeft: '24px',
      paddingRight: '24px',
      paddingTop: '12px',
    })
  })
  it('properly applies values when passed three', () => {
    expect(padding('12px', '24px', '36px')).toEqual({
      paddingBottom: '36px',
      paddingLeft: '24px',
      paddingRight: '24px',
      paddingTop: '12px',
    })
  })
  it('properly applies values when passed four', () => {
    expect(padding('12px', '24px', '36px', '48px')).toEqual({
      paddingBottom: '36px',
      paddingLeft: '48px',
      paddingRight: '24px',
      paddingTop: '12px',
    })
  })
  it('properly applies values when passed four', () => {
    expect(padding(12, 24, 36, 48)).toEqual({
      paddingBottom: 36,
      paddingLeft: 48,
      paddingRight: 24,
      paddingTop: 12,
    })
  })
  it('properly applies zero value', () => {
    expect(padding(0)).toEqual({
      paddingBottom: 0,
      paddingLeft: 0,
      paddingRight: 0,
      paddingTop: 0,
    })
  })
})
