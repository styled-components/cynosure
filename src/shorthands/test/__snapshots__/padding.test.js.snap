exports[`padding properly applies a value when passed only one 1`] = `
Object {
  "paddingBottom": "12px",
  "paddingLeft": "12px",
  "paddingRight": "12px",
  "paddingTop": "12px",
}
`;

exports[`padding properly applies values when passed four 1`] = `
Object {
  "paddingBottom": "36px",
  "paddingLeft": "48px",
  "paddingRight": "24px",
  "paddingTop": "12px",
}
`;

exports[`padding properly applies values when passed three 1`] = `
Object {
  "paddingBottom": "36px",
  "paddingLeft": "24px",
  "paddingRight": "24px",
  "paddingTop": "12px",
}
`;

exports[`padding properly applies values when passed two 1`] = `
Object {
  "paddingBottom": "12px",
  "paddingLeft": "24px",
  "paddingRight": "24px",
  "paddingTop": "12px",
}
`;
